web/profiles/custom/os2loop/modules/os2loop_flag_content/src/Form/FlagContentForm.php

```diff
@@ -0,0 +1,173 @@
+
+/**
+ * Flag content form.
+ *
+ * @package Drupal\os2loop_flag_content\Form
+ */
+class FlagContentForm extends FormBase implements ContainerInjectionInterface {
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The route mathcer.
+   *
+   * @var \Drupal\Core\Routing\RouteMatchInterface
+   */
+  protected $routeMatcher;
+
+  /**
+   * The mail manager.
+   *
+   * @var \Drupal\Core\Mail\MailManagerInterface
+   */
+  protected $mailManager;
+
+  /**
+   * The mail manager.
+   *
+   * @var \Drupal\Core\Session\AccountProxyInterface
+   */
+  protected $currentUser;
+
+  /**
+   * Config service for flag content admin settings.
+   *
+   * @var \Drupal\os2loop_flag_content\Services\ConfigService
+   */
+  protected $configService;
+
+  /**
+   * Constructs an flag content form.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
+   *   The entity type manager.
+   * @param \Drupal\Core\Routing\RouteMatchInterface $routeMatcher
+   *   The route matcher.
+   * @param \Drupal\Core\Mail\MailManagerInterface $mailManager
+   *   The mail manager.
+   * @param \Drupal\Core\Session\AccountProxyInterface $currentUser
+   *   The current user.
+   * @param \Drupal\os2loop_flag_content\Services\ConfigService $configService
+   *   The config for flag content.
+   */
+  public function __construct(EntityTypeManagerInterface $entityTypeManager, RouteMatchInterface $routeMatcher, MailManagerInterface $mailManager, AccountProxyInterface $currentUser, ConfigService $configService) {
+    $this->entityTypeManager = $entityTypeManager;
+    $this->routeMatcher = $routeMatcher;
+    $this->mailManager = $mailManager;
+    $this->currentUser = $currentUser;
+    $this->configService = $configService;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('entity_type.manager'),
+      $container->get('current_route_match'),
+      $container->get("plugin.manager.mail"),
+      $container->get("current_user"),
+      $container->get("os2loop_flag_content.config_service"),
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'flag_form_button';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $flag_config = $this->configService->getFlagContentSettings();
+    $nid = $this->routeMatcher->getRawParameter('node');
+    $form['node'] = [
+      '#type' => 'hidden',
+      '#value' => $nid,
+    ];
+
+    $config_causes = explode("\n", $flag_config->get('causes'));
+    foreach ($config_causes as $cause) {
+      $causes[$cause] = $cause;
+    }
+    $form['cause'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Causes'),
+      '#options' => $causes,
+      '#validated' => TRUE,
```

Comment:
I suppose you mean `#required` here. I think `#validated` is used only internally by the form validator.